### PR TITLE
Update Geo modules and misc modules to Drupal 10

### PR DESCRIPTION
**Pull request by weitzman**
Note: drupal/geocoder skipped due to Guzzle incompat

**Jira:** (Skip unless you are MA staff)
DP-27345

**To Test:**
- [ ] Test the modules that are mentioned in the bot comment below. At time of this writing, thats: 

1. google_map_field
2. geofield
3. prepopulate
4. scheduled_transitions
5. maxlength
6. google_tag (aka google tag manager)

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -418,9 +418,6 @@
                 "CKeditor Plugin - using a combined patch to avoid conflict (https://www.drupal.org/node/2928318)": "https://www.drupal.org/files/issues/2020-10-28/media_entity_download-ckeditor_plugin-2928318-20.patch",
                 "3106062 - Track the usage of Media entity through media_entity_download route with entity_usage": "patches/media_entity_download-track_media_entity_usage-3106062-2.patch"
             },
-            "drupal/maxlength": {
-                "Don't enforce maxlength stop, allow negative count down": "https://www.drupal.org/files/issues/2020-12-07/2916687-allow-negative-count-13.patch"
```
Comment (arthurbaghdas) [inline composer.json:422]: Why we didn't replace this patch with the latest one compatible with the 2.x version of the module, we might break something by just removing the patch.

File: conf/drupal/config/google_tag.container.default.yml [hunk 2]
```diff
@@ -8,33 +8,8 @@ dependencies:
 id: default
 label: Default
 weight: 0
-container_id: GTM-MPHNMQ
```
Comment (arthurbaghdas) [inline conf/drupal/config/google_tag.container.default.yml:11]: I"m not sure about this change, can you confirm this was exported correctly?

File: composer.json [hunk 3]
```diff
@@ -205,8 +205,8 @@
         "drupal/geofield": "^1",
         "drupal/gin": "^3",
         "drupal/gin_toolbar": "^1",
-        "drupal/google_map_field": "^1",
-        "drupal/google_tag": "^1.4",
+        "drupal/google_map_field": "^2",
```
Comment (arthurbaghdas) [inline composer.json:208]: Do we know if we are using this module?

**Review by arthurbaghdas — CHANGES_REQUESTED** [review]

File: composer.json (hunk 3 above)
Comment (weitzman) [reply composer.json:208]: I dont know. If anyone suspects that we are no longer using a module, I suggest opening a research ticket.

File: composer.json (hunk 1 above)
Comment (weitzman) [reply composer.json:422]: Its a bit hard to follow, but the second to last comment there points points to an issue which is now merged. I refer to [this comment](https://www.drupal.org/project/maxlength/issues/2916687#comment-14576442) which points to https://www.drupal.org/project/maxlength/issues/3259425. So this patch is no longer needed?

File: conf/drupal/config/google_tag.container.default.yml (hunk 2 above)
Comment (weitzman) [reply conf/drupal/config/google_tag.container.default.yml:11]: I can confirm that the update hooks ran (google_tag_update_8200, google_tag_update_8201) and the export happened after that. So technically we look OK. During our call we decided to send this PR back to @joeg8612 for deeper external review of Google Tag Manager.

File: composer.json (hunk 1 above)
Comment (arthurbaghdas) [reply composer.json:422]: After some investigation I think that we can actually remove this patch, since there is an implementation already in 2.x branch, however that requires additional changes from the UI.

**Review by arthurbaghdas — APPROVED** [review]
